src/lib/grammar.rs

```diff
@@ -386,4 +389,29 @@ mod test {
         assert_eq!(grm.prod_precs[6].unwrap(), Precedence{level: 4, kind: AssocKind::Nonassoc});
         assert!(grm.prod_precs[7].is_none());
     }
+
+    #[test]
+    fn test_prec_override() {
+        let grm = Grammar::new(&parse_yacc(&"
+            %start expr
+            %left '+' '-'
+            %left '*' '/'
+            %%
+            expr : expr '+' expr
+                 | expr '-' expr
+                 | expr '*' expr
+                 | expr '/' expr
+                 | %prec '-'  expr '*'
```

Comment:
But wouldn't `'-' expr '*'` be just a normal production that would change the language we accept? It sounds to me like that would just tell the language to accept "-5 *" as a valid program. The `%prec` is needed to say that the production `'-' expr` has the same precedence as `'*'`.